### PR TITLE
Add signal for when contents of cell are changed

### PR DESCRIPTION
**Pull request by iatenine**
Enhancement suggested by issue #11855. Creates a new signal for GridMap of "cell_contents_changed"
This signal passes info on the location of the cell updated as a Vector3

### CODE REVIEW

**Comment by aaronfranke** [comment]
@iatenine Is this still desired? If so, it needs to be rebased on the latest master branch. While there are no conflicts, rebasing is important so that reviewers can easily test the PR.

**Comment by iatenine** [comment]
> @iatenine Is this still desired? If so, it needs to be rebased on the latest master branch. While there are no conflicts, rebasing is important so that reviewers can easily test the PR.

I don't understand. I only see the single "default" master branch and some version numbers. Is the master branch I'm trying to push to incorrect?

**Comment by aaronfranke** [comment]
The branch you have selected is correct, "master" is indeed what you want.

This PR is based on top of the master branch from 6 months ago, so if anyone compiles your branch then they are compiling Godot from 6 months ago. This is why it's often good to periodically [rebase](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase). Rebasing is required if there are conflicts, but your PR has none, so rebasing is optional but still nice to do. If you don't want to, you don't have to, it's just something to make maintainers' jobs easier, since then building your branch would build a more recent version of Godot.

It's also simply good to get a reply from you so that we can see you're still active and interested in this feature.

TL;DR: You don't have to do anything if you don't want to, it's fine.

**Comment by iatenine** [comment]
Not sure if I was *meant* to create a new PR effectively but have rebased and commited here https://github.com/godotengine/godot/pull/40044 and will be therefore closing this PR
